funding/commitment_type_negotiation.go

```diff
@@ -24,38 +18,47 @@ var (
 // negotiateCommitmentType negotiates the commitment type of a newly opened
 // channel. If a desiredChanType is provided, explicit negotiation for said type
 // will be attempted if the set of both local and remote features support it.
-// Otherwise, implicit negotiation will be attempted. Two booleans are
-// returned letting the caller know if the option-scid-alias or zero-conf
-// channel types were negotiated.
+// Otherwise, implicit negotiation will be attempted.
+//
+// The returned ChannelType is nil when implicit negotiation is used. An error
+// is only returned if desiredChanType is not supported.
 func negotiateCommitmentType(desiredChanType *lnwire.ChannelType, local,
-	remote *lnwire.FeatureVector, mustBeExplicit bool) (bool,
-	*lnwire.ChannelType, lnwallet.CommitmentType, error) {
-
-	if desiredChanType != nil {
-		// If the peer does know explicit negotiation, let's attempt
-		// that now.
-		if hasFeatures(
-			local, remote, lnwire.ExplicitChannelTypeOptional,
-		) {
+	remote *lnwire.FeatureVector) (*lnwire.ChannelType,
```

Comment:
Very sorry to ask this late in the game, but coming to this fresh I still find it quite difficult to read through all the branches here. 

Would it be simpler to have something like: 
`explicitNegotiation = hasFeatures(...)`
`channelTypeRequested = desiredChannelType !=nil` 

And then just have a single switch that covers each branch that we care about? I think it could really improve the readability here.

Reply:
Switched to a switch in 81d8d31d82fcee8051ef3e0de1f531e1116ca8b0.

Reply:
I think this looks _real nice_ but of course I'm biased having asked for it 🤩  @yyforyongyu wdyt?

Reply:
yeah i like it🥳